conf/route_control.py

```diff
@@ -1,530 +1,856 @@
+                A cache to keep track of entries add in Bess.
+            _module_gate_count_cache (Dict[str, int]):
+                A cache for counting module gate occurrences.
+        """
+        self._unresolved_arp_queries_cache: Dict[str, RouteEntry] = {}
+        self._neighbor_cache: Dict[str, NeighborEntry] = {}
+        self._module_gate_count_cache: Dict[str, int] = defaultdict(lambda: 0)
+
+        self._lock = Lock()
+
+        self._ipdb = ipdb
+        self._ipr = ipr
+        self._bess_controller = bess_controller
+        self._ping_missing_thread = Thread(target=self._ping_missing_entries, daemon=True)
+        self._event_callback = None
+        self._interfaces = interfaces
+
+    def register_callbacks(self) -> None:
+        """Register callback function."""
+        logger.info("Registering netlink event listener callback...")
+        self._event_callback = self._ipdb.register_callback(self._netlink_event_listener)
+
+    def start_pinging_missing_entries(self) -> None:
+        """Starts a new thread for ping missing entries."""
+        if not self._ping_missing_thread or not self._ping_missing_thread.is_alive():
+            self._ping_missing_thread.start()
+            logger.info("Ping missing entries thread started")
+
+    def bootstrap_routes(self) -> None:
+        """Goes through all routes and handles new ones.."""
+        routes = self._ipr.get_routes()
+        for route in routes:
+            if route["event"] == KEY_NEW_ROUTE_ACTION:
+                if route_entry := self._parse_route_entry_msg(route):
+                    with self._lock:
+                        self.add_new_route_entry(route_entry)
+
+    def add_new_route_entry(self, route_entry: RouteEntry) -> None:
+        """Handles a new route entry.
+
+        Args:
+            route_entry (RouteEntry): The route entry.
+        """
+        if not (next_hop_mac := fetch_mac(self._ipdb, route_entry.next_hop_ip)):
+            logger.info(
+                "mac address of the next hop %s is not stored in ARP table. Probing...",
+                route_entry.next_hop_ip,
+            )
+            self._probe_addr(route_entry)
+            return
 
-    try:
-        ipb = ipaddress.ip_address(item.neighbor_ip)
-        if isinstance(ipb, ipaddress.IPv4Address):
-          print("The IP address {} is valid ipv4 address".format(ipb))
+        self._add_neighbor(route_entry, next_hop_mac)
+
+    def _add_neighbor(
+        self, route_entry: RouteEntry, next_hop_mac: str
+    ) -> None:
+        """Adds the route in BESS module.
+        Creates required BESS modules.
+
+        Args:
+            route_entry (RouteEntry)
+            next_hop_mac (str): The MAC address of the next hop.
+        """
+        route_module_name = self.get_route_module_name(route_entry.interface)
+        if route_entry.prefix_len == 0 and route_entry.dest_prefix == "0.0.0.0":
+            gate_idx = self.MAX_GATES - 1
```

Comment:
The issue with the pipeline for the output of the N6 interface is due to this `if` statement. This route `0.0.0.0/0` is/was already added to the `routes` module and connected to the `bad_route` module using `routes` ogate 8191 in the [ports.py](https://github.com/omec-project/upf/blob/master/conf/ports.py#L128). So, you cannot use it again the same `ogate` and this is why the output of the N6 pipeline does not get properly connected/linked.